### PR TITLE
Allow users to add host user accounts to /etc/passwd

### PR DESCRIPTION
**Pull request by rhatdan**
Some containers require certain user account(s) to exist within the
container when they are run. This option will allow callers to add a
bunch of passwd entries from the host to the container even if the
entries are not in the local /etc/passwd file on the host.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1935831

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: docs/source/markdown/podman-create.1.md
Comment (mheon) [inline docs/source/markdown/podman-create.1.md:416]: Please make it more clear that this does not set the user in the container to be the new UID or GID.

Does this work with rootless Podman, considering UID/GID mapping means the UID/GID won't actually match the one on the host? Probably need a warning about that.

Also should mention explicitly that UID is allowed.

**Comment by mheon** [comment]
Does `--hostuser=my-user --user=my-user` work? As in, creates a user identical to the one on the host, and then sets the container to start as that user?

**Comment by rhatdan** [comment]
./bin/podman run --hostuser dwalsh --user dwalsh alpine grep dwalsh /etc/passwd
Error: unable to find user dwalsh: no matching entries in passwd file

Does not work.  I will take a look.

File: docs/source/markdown/podman-create.1.md
Comment (rhatdan) [reply docs/source/markdown/podman-create.1.md:416]: Should we change the UID to match the UID on the host? within the user namespace?
IE Adding dwalsh (UID=3267) would end up as UID=0 within the container.

File: docs/source/markdown/podman-create.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:415]:
```suggestion
Add a user account to /etc/passwd from the host to the container. The Username or UID must```

File: docs/source/markdown/podman-run.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:452]: If you take above, do the same here.

**Comment by rhatdan** [comment]
@giuseppe WDYT, should the UIDs be modified based on the user namespace within the container?

File: libpod/container_internal_linux.go
Comment (cdoern) [inline libpod/container_internal_linux.go:2326]: maybe make a function for this since done a few times. similar to https://github.com/containers/podman/blob/dbdf4761abb6d68169c8fdd24bd0c44d38c1c5f8/pkg/domain/infra/abi/images.go#L839-L844 once this merges you can use this

**Comment by giuseppe** [comment]
I think we shouldn't do any conversion to the ids in the namespace. 
The rootless user won't anyway be able to use IDs outside of the allocated range and the ID should be relative to the container image so I think it is correct without converting it.

File: test/system/030-run.bats
Comment (giuseppe) [inline test/system/030-run.bats:715]: why does it need rootless?

Comment (rhatdan) [reply test/system/030-run.bats:715]: Mainly because I am not sure how to figure out a user which is not in the container to add to the container.  If you are running rootless, I can add the current user.

If running as root, I could add a user to the system and then use that, but I am not crazy about modifying a system when users run the tests.

@edsantiago Any ideas?

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

File: docs/source/markdown/podman-create.1.md
Comment (vrothberg) [inline docs/source/markdown/podman-create.1.md:416]:
```suggestion
Add a user account to /etc/passwd from the host to the container. The username
or UID must exist on the host system.
```
Same below.

**Pull request merged**
